pkg/downloader/chart_downloader.go

```diff
@@ -365,34 +368,70 @@ func pickChartRepositoryConfigByName(name string, cfgs []*repo.Entry) (*repo.Ent
 // Charts are not required to be included in an index before they are valid. So
 // be mindful of this case.
 //
-// The same URL can technically exist in two or more repositories. This algorithm
-// will return the first one it finds. Order is determined by the order of repositories
-// in the repositories.yaml file.
+// The same URL can technically exist in multiple repositories.
+// This algorithm will return one of them based on concurrent processing,
+// without regard to the order specified in the repositories.yaml file.
```

Comment:
minor, it isn't necessary to allude to the functions prior behavior

```suggestion
// All repositories in repositories.yaml are searched concurrently, and the first found returned
```